File: Hard/2026-01-03-1411-Number-of-Ways-to-Paint-N-3-Grid/solution.java
class Solution {
    public int numOfWays(int n) {
        final int MOD = 1000000007;
        long same = 6; // Ways to paint two adjacent cells the same color
        long diff = 6; // Ways to paint two adjacent cells different colors
        long total = (same + diff) % MOD;

        for (int i = 2; i <= n; i++) {
            long newSame = (diff * 2) % MOD;
            long newDiff = (total * 3) % MOD;
            same = newSame;
            diff = newDiff;
            total = (same + diff) % MOD;
        }

        return (int) total;
    }
}

File: Hard/2026-01-03-1411-Number-of-Ways-to-Paint-N-3-Grid/solution.js
var numOfWays = function(n) {
    const MOD = 1e9 + 7;
    let same = 6; // Ways to paint two adjacent cells the same color
    let diff = 6; // Ways to paint two adjacent cells different colors

    for (let i = 2; i <= n; i++) {
        let newSame = (diff * 2) % MOD;
        let newDiff = (same + diff * 2) % MOD;
        same = newSame;
        diff = newDiff;
    }

    return (same + diff) % MOD;
};